django/db/backends/sqlite3/operations.py

```diff
@@ -33,15 +33,21 @@ def bulk_batch_size(self, fields, objs):
         limit = 999 if len(fields) > 1 else 500
         return (limit // len(fields)) if len(fields) > 0 else len(objs)
 
-    def check_aggregate_support(self, aggregate):
+    def check_expression_support(self, expression):
         bad_fields = (fields.DateField, fields.DateTimeField, fields.TimeField)
-        bad_aggregates = (aggregates.Sum, aggregates.Avg,
-                          aggregates.Variance, aggregates.StdDev)
-        if aggregate.refs_field(bad_aggregates, bad_fields):
-            raise NotImplementedError(
-                'You cannot use Sum, Avg, StdDev and Variance aggregations '
-                'on date/time fields in sqlite3 '
-                'since date/time is saved as text.')
+        bad_aggregates = (aggregates.Sum, aggregates.Avg, aggregates.Variance, aggregates.StdDev)
+        try:
+            if isinstance(expression, bad_aggregates) and isinstance(expression.input_field.output_field, bad_fields):
```

Comment:
I find this code a bit confusing -- where is the `FieldError` expected to be raised? `expression.input_field.output_field`? Could you use an if statement for that check rather than try/except or maybe move the code that's not expected to raise into an else block of this try/except? .Not an expert here, so maybe it's fine as is.